### ISSUE TITLE
Communication failure between nodes inside a cluster when 3 replicas are configured

### ISSUE THREAD

**Issue by AsT4re** [dgraph-io/dgraph#1617]
I have a cluster of 3 nodes deployed with latest docker image (v0.8.3):

Node 1 (1 dgraphzero instance + 1 dgraph instance):
`docker run -it -p 8080:8080 -p 9080:9080 -p 8888:8888 -p 12345:12345 -v ~/dgraph:/dgraph --name dgraph dgraph/dgraph:latest dgraphzero -w zw --bindall=true --my "<IpNode1>:8888" --replicas 3`

`docker exec -it dgraph dgraph --idx 1 --peer "127.0.0.1:8888" --my "<IpNode1>:12345" --bindall=true --memory_mb=8192`

Node 2 (1 dgraph instance):
`docker run -it -p 8080:8080 -p 9080:9080 -p 12345:12345 -v ~/dgraph:/dgraph --name dgraph dgraph/dgraph:latest dgraph --idx 2 --peer "<IpNode1>:8888" --my "<IpNode2>:12345" --bindall=true --memory_mb=8192`

Node 3 (1 dgraph instance):
`docker run -it -p 8080:8080 -p 9080:9080 -p 12345:12345 -v ~/dgraph:/dgraph --name dgraph dgraph/dgraph:latest dgraph --idx 3 --peer "<IpNode1>:8888" --my "<IpNode3>:12345" --bindall=true --memory_mb=8192`

I am sending lots of datas with Go client BatchSet() API to Node 1 but very quickly the data stops being added on Node 1 and Node 1 keeps sending messages to Node 2 and Node 3 for replicating the data ("SENDING: MsgApp 1-->2" and "SENDING: MsgApp 1-->3") but Node 2 and Node 3 doesn't seem to receive these messages, normally I see this in logs: "RECEIVED: MsgApp 1-->2" + "SENDING: MsgAppResp 2-->1", here nothing is displayed and there is no data added to 2 and 3.

Using exactly the same configuration with 2 nodes and replicas option set to "2", everything seems to work well.
I have also tested configurations with different servers and all the nodes seems to be connected correctly between each others.

Thank you for your help.

**Comment by pawanrawal** [comment]
This seems to work fine when using the Dgraph binaries directly. Trying to reproduce it with Docker now.

**Comment by AsT4re** [comment]
Hi pawanrawal, 

so after investigate a little bit more about my issue I figured out that my scenario wasn't exactly right.

The bug occure when there are 3 dgraph instances running on 3 different servers (3 different ips) with a dgraphzero instance running with "replica" option set to "3" and when all mutations requests are sent on **the node elected as leader**

It works in these scenarios:
- Requests are sent on an another node than the leader on cluster with 3 dgraph instances running on 3 different servers.
- Requests are sent on the leader on cluster with 2 dgraph instances running on 2 different servers.

I did not made any tests on a clusters with more than 3 nodes.
